crates/export_ts/src/main.rs

```diff
@@ -0,0 +1,85 @@
+use std::env::args;
+use std::fs::write;
+use std::path::PathBuf;
+#[allow(unused_imports)]
+use ts_rs::TS;
+
+#[allow(dead_code)]
+const TS_RS_PREAMBLE_STR: &str = "// This file was generated by [ts-rs](https://github.com/Aleph-Alpha/ts-rs). Do not edit this file manually.\n\n";
+const CUSTOM_PREAMBLE_STR: &str = "// This file was generated by [`export_ts` from the Holochain project](https://github.com/holochain/holochain/tree/main/crates/export_ts). Do not edit this file manually.\n\n";
+
+macro_rules! fill_and_save_buffer {
+    ($buffer_name: ident, $path: expr, $file_name: expr, $code: block) => {
+        // TODO: remove this override once future usage starts writing to
+        // a buffer.
+        #[allow(unused_mut)]
+        let mut $buffer_name = String::new();
+        $buffer_name.push_str(CUSTOM_PREAMBLE_STR);
+        let file_path = $path.join($file_name);
+        print!("Populating file '{}'...\n\n", file_path.display());
+        $code
+        // Write all the types we've collected into the file.
+        write(
+            file_path,
+            $buffer_name.as_str(),
+        )?;
+        print!("Done!\n\n");
+    };
+}
+
+#[allow(unused_macros)]
+macro_rules! add_type_to_buffer {
+    ($type: ty, $buf: ident, $inline_comment: expr) => {
+        print!("Exporting {}... ", stringify!($type));
+        let result = <$type as TS>::export_to_string();
```

Comment:
Same here, it looks like you don't need the whole code inside of the macro. It's ideal to have as little code as possible in a macro.